addons/maptools/CfgVehicles.hpp

```diff
@@ -52,7 +52,7 @@ class CfgVehicles {
                 };
                 class ACE_MapToolsAlignCompass {
                     displayName = CSTRING(MapToolsAlignCompass);
-                    condition = QUOTE((GVAR(mapTool_Shown) != 0) && {'ItemCompass' in assigneditems ACE_player});
+                    condition = QUOTE((GVAR(mapTool_Shown) != 0) && {getUnitLoadout ACE_player param [9, []] param [3, ''] != ''});
```

Comment:
don't we need the `ARR_x` macros

Reply:
I can update this, but I'm not 100% where I should use the macro

Is this correct?
`getUnitLoadout ACE_player param ARR_2(9,[]) param ARR_2(3,'')`

Do I do anything about the `[]` default value?

Reply:
> Is this correct? `getUnitLoadout ACE_player param ARR_2(9,[]) param ARR_2(3,'')`

Well it is definitely better because with this change the build runs successfully on my machine. Before it didn't. 

> Do I do anything about the `[]` default value?

Do you mean wether you should do another substitution with something like `ARR_0` (which doesn't exist) as opposed to `[]` ?

Reply:
> > Is this correct? `getUnitLoadout ACE_player param ARR_2(9,[]) param ARR_2(3,'')`
> 
> Well it is definitely better because with this change the build runs successfully on my machine. Before it didn't.
> 
> > Do I do anything about the `[]` default value?
> 
> Do you mean wether you should do another substitution with something like `ARR_0` (which doesn't exist) as opposed to `[]` ?

That's what was confusing me yeah, it's an array, but it's an empty array. ARR macros replace square brackets so do I need to replace those with something

Reply:
> ARR macros replace square brackets so do I need to replace those with something

As far as I understand, the ARR_x macros do **not** replace the square brackets (see [here](https://github.com/acemod/ACE3/blob/master/include/x/cba/addons/main/script_macros_common.hpp#L1066-L1084)). 
They only preserve the commas that would get erased by the other macro `QUOTE` (see https://community.bistudio.com/wiki/PreProcessor_Commands#Arguments). 
Hence the code wouldn't compile without the ARR_x macros.

My takeaway from this is that the square brackets are still needed (also _around_ the ARR_ macros, not just _within_ them).
`getUnitLoadout ACE_player param [ARR_2(9,[])] param [ARR_2(3,'')] != ''`

Can anyone with more experience in SQF confirm this?
(I am just extrapolating from my C/C++ background. But I am not good with SQF and potential differences between its preprocessor and that from C/C++.)

Reply:
```suggestion
                    condition = 'GVAR(mapTool_Shown) != 0 && {getUnitLoadout ACE_player param [9, []] param [3, ""] != ""}';
```
There you go, stop worrying about `ARR_n`. :P

Reply:
Why is code always so hard to grasp as soon as you start using macros? 😅

Reply:
Because there is a good use of macros, that help with naming, etc., and there is a bad use, that needlessly complicates things.